src/middlewared/middlewared/plugins/account.py

```diff
@@ -268,32 +267,53 @@ class Config:
         if (
             'home' in data and
             data['home'] not in (user['home'], '/nonexistent') and
-            not data["home"].startswith(f'{user["home"]}/')
+            not data['home'].startswith(f'{user["home"]}/')
         ):
             home_copy = True
             home_old = user['home']
         else:
             home_copy = False
 
+        # After this point user dict has values from data
         user.update(data)
 
-        password = await self.__set_password(user)
-
-        await self.__update_sshpubkey(user, group['bsdgrp_group'])
-        user.pop('sshpubkey', None)
+        if home_copy and not os.path.isdir(user['home']):
+            try:
+                os.makedirs(user['home'])
+                os.chown(user['home'], user['uid'], group['bsdgrp_gid'])
+            except OSError:
+                self.logger.warn('Failed to chown homedir', exc_info=True)
+            if not os.path.isdir(user['home']):
+                raise CallError(f'{user["home"]} is not a directory')
 
         home_mode = user.pop('home_mode', None)
-        if home_mode is not None:
-            if not user['builtin'] and os.path.exists(user['home']):
+        if user['builtin']:
+            home_mode = None
+
+        def set_home_mode():
+            if home_mode is not None:
                 try:
                     os.chmod(user['home'], int(home_mode, 8))
                 except OSError:
                     self.logger.warn('Failed to set homedir mode', exc_info=True)
 
         if home_copy:
+            await self.__update_sshpubkey(home_old, user, group['bsdgrp_group'])
+
             def do_home_copy():
-                subprocess.run(f"su - {user['username']} -c '/bin/cp -a {home_old}/* {user['home']}/'")
+                try:
+                    subprocess.run(f"/usr/bin/su - {user['username']} -c '/bin/cp -a {home_old}/ {user['home']}/'", shell=True, check=True)
+                except subprocess.CalledProcessError as e:
+                    self.logger.warn(f"Failed to copy homedir: {e}")
+                set_home_mode()
```

Comment:
Why does this has to be set later? The dir already exists, and its not recursive.

Reply:
Didn't know that! Thanks for clarifying!